packages/core/ui-settings/core-ui-settings-server-internal/src/routes/delete.ts

```diff
@@ -51,15 +51,33 @@ export function registerDeleteRoute(router: InternalUiSettingsRouter) {
       throw error;
     }
   };
+  if (publicApiEnabled) {
```

Comment:
General comment for all routes:

Either we decide that for now, those two API prefix (/api and /internal) should have **the exact same behavior**, in which case I would reuse the same handler, instead of duplicating them, e.g


```ts
const deleteHandler: HttpRequestHandler<forgotWhichGenerics> = async (context, request, response) => {
        const uiSettingsClient = (await context.core).uiSettings.client;
        return await deleteFromRequest(uiSettingsClient, context, request, response);
      }

if (publicApiEnabled) {
   router.delete({ path: '/api/kibana/settings/{key}', validate }, deleteHandler);
}
router.delete({ path: '/internal/kibana/global_settings/{key}', validate, options: { access: 'internal' } }, deleteHandler)

```

Or we decide that the code may diverge, in which case the registration should be fully dissociated, e.g with a dedicated file for the internal and the public counterpart of each API.

Reply:
I agree that separating the public and the internal routes in different files is the better option given what we discussed about potentially evolving the public routes in the future.
I took the Core routes approach for structuring the files (having public routes in `/src/routes` and internal ones in `/src/routes/internal`).